test/AutoDiff/differentiable_func_type_type_checking.swift

```diff
@@ -63,3 +63,10 @@ let diffVectorFunc: (Vector<Float>) -> Vector<Float>
 inferredConformancesGeneric(diffVectorFunc) // okay!
 
 func inferredConformancesGenericResult<T, U>() -> @differentiable (Vector<T>) -> Vector<U> {}
+
+//
+// linear function type
+//
+
+// expected-error @+1 {{cannot mark types as linear differentiable}}
+let _: @differentiable(linear) (Float) -> Float
```

Comment:
I'd suggest adding more ambiguity tests because parsing is tricky. See https://github.com/rxwei/swift/blob/ffc1370297eb2903b3a6ab2254c84c8ecf616bf1/test/AutoDiff/differentiable_func_type_parse.swift.

Reply:
I meant to attach this comment to the parsing test file.

Reply:
Similar to my comment above regarding adding the back tracking, I'm now better able to handle these test cases so I added several of them. 😄